readthedocs/projects/models.py

```diff
@@ -541,13 +555,117 @@ def get_production_media_url(self, type_, version_slug):
 
         if self.is_subproject:
             # docs.example.com/_/downloads/<alias>/<lang>/<ver>/pdf/
-            path = f'//{domain}/{DOC_PATH_PREFIX}downloads/{self.alias}/{self.language}/{version_slug}/{type_}/'  # noqa
+            path = f'//{domain}/{self.proxied_api_url}downloads/{self.alias}/{self.language}/{version_slug}/{type_}/'  # noqa
         else:
             # docs.example.com/_/downloads/<lang>/<ver>/pdf/
-            path = f'//{domain}/{DOC_PATH_PREFIX}downloads/{self.language}/{version_slug}/{type_}/'
+            path = f'//{domain}/{self.proxied_api_url}downloads/{self.language}/{version_slug}/{type_}/'  # noqa
 
         return path
 
+    @property
+    def proxied_api_host(self):
+        """
+        Used for the proxied_api_host in javascript.
+
+        This needs to start with a slash at the root of the domain,
+        and end without a slash
+        """
+        if self.urlconf:
+            # Add our proxied api host at the first place we have a $variable
+            # This supports both subpaths & normal root hosting
+            url_prefix = self.urlconf.split('$', 1)[0]
+            return '/' + url_prefix.strip('/') + '/_'
+        return '/_'
+
+    @property
+    def proxied_api_url(self):
+        """
+        Like the api_host but for use as a URL prefix.
+
+        It can't start with a /, but has to end with one.
+        """
+        return self.proxied_api_host.strip('/') + '/'
+
+    @property
+    def real_urlconf(self):
```

Comment:
The point here is that it's converted into a regex, instead of being the user-facing simplified version. I don't think `user_urlconf` conveys that at all.

Reply:
I'm reading `user_urlconf` as the urlconf given by the user (`/$language/$version/`) and `django_urlconf` as an already user urlconf parsed and converted to what django expects.

Reply:
I think `regex_urlconf` works reasonably well.